### PR TITLE
Feature/basic test support

### PR DESCRIPTION
**Pull request by nvdk**
~there's still an issue with the template I'd like to solve (looks like the shell doesn't quit after running the tests), but seems to be working well otherwise.~

this is resolved now

### CODE REVIEW

**Comment by nvdk** [comment]
this turned into quite a refactor, apologies

File: models/decision.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import {analyse} from '@lblod/marawa/rdfa-context-scanner';
+import {cleanupTriples} from '../support/pre-importer';
+import { editorDocumentFromUuid } from '../support/editor-document';
+
+export default class Decision {
+  // TODO: perhaps not the best name or place, given that it returns an array
```
Comment (lagartoverde) [inline models/decision.js:6]: Maybe a helper function like extractDecisionsFromDocument is better?

File: models/vote.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+       SELECT DISTINCT * WHERE {
+         ${sparqlEscapeUri(treatment)} besluit:heeftStemming ?uri.
+         OPTIONAL { ?uri besluit:onderwerp ?subject }
+         OPTIONAL { ?uri besluit:gevolg ?result. }
+      }
+  `);
+    return result.results.bindings.map((binding) => Vote.fromBinding(binding));
+  }
+
+  static fromBinding({uri, subject, result}) {
```
Comment (lagartoverde) [inline models/vote.js:17]: I really like all this from binding methods

File: support/agenda-utils.js [hunk 3]
```diff
@@ -19,15 +17,19 @@ async function getDataForAgenda(meetingUuid, agendaKindUuid) {
...
-        agendapoint.type = agendaType.uri;
-        agendapoint.typeName = agendaType.label;
-      }
-    }
+    ensureAgendapointType(agendapoints, agendaType);
   }
   return { meeting, agendapoints };
 }
+
+export function ensureAgendapointType(agendapoints, type) {
```
Comment (lagartoverde) [inline support/agenda-utils.js:25]: I don't understand the name of this function, why ensure instead of something like assign?

**Review by lagartoverde — APPROVED** [review]
I added minor suggestions but overall good job is much cleaner and I think we needed an structure like this 👍

File: support/agenda-utils.js (hunk 3 above)
Comment (abeforgit) [reply support/agenda-utils.js:25]: cause we don't overwrite if it already has one. Assign would imply overwriting, and I think we use ensure in other places to signify that we only create/add information if it's not already there

File: models/decision.js (hunk 1 above)
Comment (abeforgit) [reply models/decision.js:6]: yeah that name sounds good
could stay as a static function here, don't think it's too out of place

File: support/agenda-utils.js (hunk 3 above)
Comment (lagartoverde) [reply support/agenda-utils.js:25]: But when it won't have one? I think we are fetching then without one and then adding them

File: support/agenda-utils.js [hunk 4]
```diff
@@ -19,15 +17,19 @@ async function getDataForAgenda(meetingUuid, agendaKindUuid) {
...
+  for (const agendapoint of agendapoints) {
+    if (! agendapoint.type) {
+      agendapoint.type = type.uri;
+      agendapoint.typeName = type.label;
+    }
+  }
+}
 /**
```
Comment (abeforgit) [inline support/agenda-utils.js:33]: this comment is no longer accurate

File: test/agenda-publication-test.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
+import  { strict  as assert, equal } from 'assert';
```
Comment (abeforgit) [inline test/agenda-publication-test.js:1]: Maybe add a line in the readme on how to run these?

**Review by abeforgit — APPROVED** [review]
Great work!

File: support/agenda-utils.js (hunk 3 above)
Comment (nvdk) [reply support/agenda-utils.js:25]: We fetch it optionally, so indeed we only set if it's not set yet. e.g. if you haven't published any agenda yet.
e.g. :
- you publish the planned agenda: all agendapoints get the type planned
- you add some agendapoints
- you publish the additional agenda this contains the earlier agendapoints with type planned and the new ones with type "additional"

Comment (lagartoverde) [reply support/agenda-utils.js:25]: Ohhh indeed, then it makes sense thanks for explaining :)

**Pull request merged**